test/test_kernels/test_aggressive_momentum_model_free_kernel.py

```diff
@@ -0,0 +1,331 @@
+def test_aggressive_momentum_first_kernel_step(
+    mock_compute_parameter_update_loss: Mock,
+    mock_increase_momentum_parameters: Mock,
+    mock_accept_matrix_updates: Mock,
+    mock_decrease_momentum_parameters: Mock,
+    mock_reject_matrix_updates: Mock,
+    mock_construct: Mock,
+    mock_apply_momentum: Mock,
+    mock_find_low_rank: Mock,
+    mock_compute_loss: Mock,
+    fixture_with_tol: Fixture,
+) -> None:
+    (indata, _, kernel) = fixture_with_tol
+    shape_X = indata.sparse_matrix_X.shape
+    mock_construct.return_value = np.full(shape_X, 4.0)
+    mock_apply_momentum.return_value = np.full(shape_X, 5.0)
+    mock_find_low_rank.return_value = np.full(shape_X, 6.0)
+    mock_compute_loss.return_value = 3.0
+
+    initial_utility_Z = kernel.utility_matrix_Z
+
+    kernel.step()
+    mock_construct.assert_called_once_with(
+        indata.low_rank_candidate_L, indata.sparse_matrix_X
+    )
+    mock_apply_momentum.assert_called_once_with(
+        mock_construct.return_value,
+        initial_utility_Z,
+        kernel.momentum_beta,
+    )
+    mock_find_low_rank.assert_called_once_with(
+        mock_apply_momentum.return_value,
+        kernel.target_rank,
+        indata.low_rank_candidate_L,
+        kernel.svd_strategy,
+    )
+
+    mock_compute_loss.assert_called_once_with(
+        mock_apply_momentum.return_value,
+        mock_find_low_rank.return_value,
+        LossType.FROBENIUS,
+    )
+
+    mock_compute_parameter_update_loss.assert_not_called()
+    mock_increase_momentum_parameters.assert_not_called()
+    mock_accept_matrix_updates.assert_not_called()
+    mock_decrease_momentum_parameters.assert_not_called()
+    mock_reject_matrix_updates.assert_not_called()
+
+
+@patch(f"{PKG}.compute_loss")
+@patch(f"{PKG}.reconstruct_X_from_L")
+def test_compute_parameter_update_loss(
+    mock_reconstruct_X_from_L: Mock,
+    mock_compute_loss: Mock,
+    fixture_with_tol: Fixture,
+) -> None:
+    (_, _, kernel) = fixture_with_tol
+    kernel.parameter_update_loss = 1.0
+    kernel.elapsed_iterations = 1
+
+    mock_reconstruct_X_from_L.return_value = np.eye(3)
+
+    kernel.compute_parameter_update_loss()
+    mock_reconstruct_X_from_L.assert_called_once_with(kernel.low_rank_candidate_L)
+    mock_compute_loss.assert_called_once_with(
+        kernel.sparse_matrix_X,
+        mock_reconstruct_X_from_L.return_value,
+        LossType.FROBENIUS,
+    )
+
+
+@patch(f"{PKG}.{KERNEL_CLASS}.compute_parameter_update_loss")
+def test_compute_parameter_update_loss_correct_assignment(
+    mock_compute_parameter_update_loss: Mock,
+    fixture_with_tol: Fixture,
+) -> None:
+    (_, _, kernel) = fixture_with_tol
+    kernel.parameter_update_loss = 1.0
+    kernel.elapsed_iterations = 1
+    mock_compute_parameter_update_loss.return_value = 3.0
+
+    kernel.step()
+    mock_compute_parameter_update_loss.assert_called_once()
+    assert kernel.parameter_update_loss == 3.0
```

Comment:
Might be slightly more robust to refer to the variable, i.e. `assert kernel.parameter_update_loss == mock_compute_parameter_update_loss.return_value`